spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaConsumerProperties.java

```diff
@@ -253,12 +255,37 @@ public void setDestinationIsPattern(boolean destinationIsPattern) {
 		this.destinationIsPattern = destinationIsPattern;
 	}
 
+	/**
+	 * No longer used; get properties such as this via {@link #getTopic()}.
+	 * @return Kafka admin properties
+	 * @deprecated No longer used
+	 */
+	@Deprecated
+	@DeprecatedConfigurationProperty(reason = "Not used since 2.1.1, set properties such as this via 'topic'")
+	@SuppressWarnings("deprecation")
 	public KafkaAdminProperties getAdmin() {
-		return this.admin;
+		final KafkaAdminProperties kafkaAdminProperties = new KafkaAdminProperties();
+		kafkaAdminProperties.setReplicationFactor(this.topic.getReplicationFactor());
+		kafkaAdminProperties.setReplicasAssignments(this.topic.getReplicasAssignments());
+		kafkaAdminProperties.setConfiguration(this.topic.getProperties());
+		return kafkaAdminProperties;
 	}
 
 	public void setAdmin(KafkaAdminProperties admin) {
-		this.admin = admin;
+		// Temporary workaround to copy the admin properties to the topic one. Not needed once the 'admin' field will be removed
+		final KafkaTopicProperties kafkaTopicProperties = new KafkaTopicProperties();
```

Comment:
This is no longer needed; just `this.topic = admin;` (since admin is a subclass).